core/trino-main/src/main/java/io/trino/execution/CreateViewTask.java

```diff
@@ -76,6 +80,17 @@ public String getName()
 
         accessControl.checkCanCreateView(session.toSecurityContext(), name);
 
+        if (metadata.isView(session, name) && !statement.isReplace()) {
+            throw semanticException(ALREADY_EXISTS, statement, "View '%s' already exists", name);
+        }
+        if (metadata.isMaterializedView(session, name)) {
+            throw semanticException(ALREADY_EXISTS, statement, "Materialized view '%s' already exists", name);
+        }
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, name);
+        if (tableHandle.isPresent()) {
```

Comment:
sorry, @lukasz-stec . I don't understand what you mean, , can you elaborate more？

Reply:
Sorry, I should say *may* break...

So because `getTableHandle` will return not empty `TableHandle` for a view in hive connector, when `statement.isReplace() == true`  (i.e. `CREATE OR REPLACE VIEW` was used) the method will trow
`throw semanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", name);` and it should replace the view definition successfuly.

Reply:
Thank you for your reminder, I still need to check if it is view when `statement.isReplace() == true`.Sorry ,can @lukasz-stec  you review again.